splice_machine/src/main/java/com/splicemachine/derby/stream/utils/ExternalTableUtils.java

```diff
@@ -64,15 +66,18 @@ public static StructType supportAvroDateType(StructType schema, String storedAs)
         return schema;
     }
 
-    public static void supportAvroDateTypeColumns(ExecRow execRow) throws StandardException {
-        for(int i=0; i < execRow.size(); i++){
-            if (execRow.getColumn(i + 1).getTypeName().equals("DATE")) {
-                execRow.setColumn(i + 1, new SQLVarchar());
+    public static Dataset<Row> castDateTypeInAvroDataSet(Dataset<Row> dataset, StructType tableSchema) {
+        int i = 0;
+        for (StructField sf : tableSchema.fields()) {
+            if (sf.dataType().sameType(DataTypes.DateType)) {
+                String colName = dataset.schema().fields()[i].name();
+                dataset = dataset.withColumn(colName, dataset.col(colName).cast(DataTypes.DateType));
             }
+            i++;
         }
+        return dataset;
     }
```

Comment:
Is supportAvroDateTypeColumns() function effectively deleted? And, it seems castDateTypeInAvroDataSet() is used only once. Is it expected to be used in other places, too?

Reply:
`supportAvroDateTypeColumns` wasn't used anywhere. yes `castDateTypeInAvroDataSet` is only used in one place, but i put it next to `supportAvroDateType`, because these two functions are related. That's why it's in ExternalTableUtils.

Reply:
OK, that makes sense. Thanks!